pkg/e2e/framework.go

```diff
@@ -191,11 +194,32 @@ func (c *E2eCLI) RunCmd(args ...string) *icmd.Result {
 
 // RunDockerCmd runs a docker command, expects no error and returns a result
 func (c *E2eCLI) RunDockerCmd(args ...string) *icmd.Result {
+	if len(args) > 0 && args[0] == compose.PluginName {
+		c.test.Fatal("This test called 'RunDockerCmd' for 'compose'. Please prefer 'RunDockerComposeCmd' to be able to test as a plugin and standalone")
+	}
 	res := c.RunDockerOrExitError(args...)
 	res.Assert(c.test, icmd.Success)
 	return res
 }
 
+const composeStandaloneEnvVar = "COMPOSE_STANDALONE_MODE"
+
+// RunDockerComposeCmd runs a docker compose command, expects no error and returns a result
+func (c *E2eCLI) RunDockerComposeCmd(args ...string) *icmd.Result {
+	composeStandaloneMode := os.Getenv(composeStandaloneEnvVar)
+	if composeStandaloneMode == "true" || composeStandaloneMode == "1" {
```

Comment:
```suggestion
composeStandaloneMode := false

// RunDockerComposeCmd runs a docker compose command, expects no error and returns a result
func (c *E2eCLI) RunDockerComposeCmd(args ...string) *icmd.Result {
	if composeStandaloneMode {
```
then in the main:
```
func TestMain(m *testing.M) {
	logrus.Info("Running tests on COMPOSE_STANDALONE_MODE=true")
	composeStandaloneMode = true
	exitCode := m.Run()
	if exitCode != 0 {
		os.Exit(exitCode)
	}
	logrus.Info("Running tests on COMPOSE_STANDALONE_MODE=false")
	composeStandaloneMode = false
	os.Exit(m.Run())
}
```
We don’t really need an env var, do we?

Reply:
would be nice this can be set as a `go test` argument so that we can parallelize run on CI

Reply:
Not sure if parallelising this would be possible... 
The e2e tests open ports and maybe that would conflict with it's twin test

Reply:
Also, I don't see a good way to run one specific test (by passing `E2E_TEST` environment variable) on one specific mode and not having to replicate Makefile targets or something like that...

Reply:
this is what I had in mind: two targets, two parallel GHA workfows
never mind

Reply:
What’s a Makefile? 😬 

It’s already possible to focus tests out of the box using `-run` with `go test`: https://pkg.go.dev/testing#hdr-Subtests_and_Sub_benchmarks
(I don’t have make on my machines)

Reply:
@mat007 Yep. And that's what the Makefile does. The thing is, when targeting a specific test `TestMain` doesn't run. So that would execute just one of the modes (the default one).
Then the question is... Do we separate the execution of the 2 modes into different runs or leave it in the same (like it's implemented in this PR).
Now I'm leaning to having 2 runs in the GHA as @ndeloof said cuz that would be easier to run in local. Cuz I don't want to run the 2 things all the time I test locally. But for the CI that's important. And also that can be run in parallel...